store/mockstore/mocktikv/cluster.go

```diff
@@ -269,6 +270,49 @@ func (c *Cluster) GetRegionByID(regionID uint64) (*metapb.Region, *metapb.Peer)
 	return nil, nil
 }
 
+// ScanRegions returns at most `limit` regions from given `key` and their leaders.
+func (c *Cluster) ScanRegions(key []byte, limit int) ([]*metapb.Region, []*metapb.Peer) {
+	c.RLock()
+	defer c.RUnlock()
+
+	regions := make([]*Region, 0, len(c.regions))
+	for _, region := range c.regions {
```

Comment:
You can `copy(regions, c.regions)` directly.

Reply:
`c.regions` is a map and I only needs its value. It seems that `copy` doesn't work well here.